src/main/java/com/cozyhome/onlineshop/orderservice/service/builder/DeliveryBuilder.java

```diff
@@ -0,0 +1,57 @@
+package com.cozyhome.onlineshop.orderservice.service.builder;
+
+import com.cozyhome.onlineshop.dto.order.DeliveryDto;
+import com.cozyhome.onlineshop.exception.DataNotFoundException;
+import com.cozyhome.onlineshop.orderservice.model.AddressDelivery;
+import com.cozyhome.onlineshop.orderservice.model.Delivery;
+import com.cozyhome.onlineshop.orderservice.model.DeliveryCompany;
+import com.cozyhome.onlineshop.orderservice.model.PostalDelivery;
+import com.cozyhome.onlineshop.orderservice.model.enums.EntityStatus;
+import com.cozyhome.onlineshop.orderservice.repository.DeliveryCompanyRepository;
+import com.cozyhome.onlineshop.orderservice.repository.DeliveryRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class DeliveryBuilder {
+    private final DeliveryCompanyRepository deliveryCompanyRepository;
+    private final DeliveryRepository deliveryRepository;
+
+    public Delivery buildPostalDelivery(DeliveryDto deliveryDto) {
+        DeliveryCompany deliveryCompany = deliveryCompanyRepository.findByNameAndStatus(deliveryDto.getDeliveryCompanyName(),
+                EntityStatus.ACTIVE)
+                .orElseThrow(() -> new DataNotFoundException(String.format("Delivery company %s doesn't found", deliveryDto.getDeliveryCompanyName())));
+
+        Delivery delivery = PostalDelivery.builder()
+                .city(deliveryDto.getCity())
+                .paymentMethod(deliveryDto.getPaymentMethod())
+                .createdAt(LocalDateTime.now())
+                .modifiedAt(LocalDateTime.now())
```

Comment:
should we fill this field during creating a delivery?

Reply:
done